k2/python/csrc/torch/fsa_algo.cu

```diff
@@ -157,6 +158,37 @@ static void PybindArcSort(py::module &m) {
       py::arg("src"), py::arg("need_arc_map") = true);
 }
 
+static void PybindShortestDistance(py::module &m) {
+  m.def(
```

Comment:
I'm concerned about how we'll extend this to the FsaVec case, given that the interfaces have to be different.
Perhaps we could call them {fsa_shortest_distance,fsavec_shortest_distance} and so on?

Reply:
I think Fsa and FsaVec can share the same interface.

If we assume that there is only one best path, then what we need in Python
is just the arc indexes of the best path. If it is a FsaVec, the arc indexes would
be idx012. On the Python side, we would use `score.index_select(0, arc_indexes).sum()`
to compute the distance.

Reply:
No, because the natural output of ShortestDistance for a vector of FSAs is a Tensor of distances of shape (num_fsas,), and for that we need to know which distances come from which FSAs.

Reply:
> natural output of ShortestDistance for a vector of FSAs is a Tensor of distances of shape (num_fsas,)

But this tensor cannot help backprop. For backprop, I think only the arc indexes
matter (for tropical semiring). We can compute the distances from the returned arc_indexes.

Reply:
For this particular application it doesn't matter because we just want to sum the result, but the point is we are creating a general-purpose interface, and I don't want to have to change it more than necessary.